eng/common/scripts/Package-Properties.ps1

```diff
@@ -78,78 +78,68 @@ function Get-PkgProperties
+            $ServiceDirectoryName = $ServiceDirectory
         }
+        $AllPkgProps = Get-AllPkgProperties -ServiceDirectoryName $ServiceDirectoryName
+    }
+    else 
+    {
+        $AllPkgProps = Get-AllPkgProperties
+    }
 
-        if ($pkgProps -ne $null)
+    foreach ($pkgProp in $AllPkgProps)
+    {
+        if(($pkgProp.Name -eq $PackageName) -or ($pkgProp.ArtifactName -eq $PackageName))
         {
-            return $pkgProps
+            return $pkgProp
         }
     }
-    LogWarning "Failed to retrive Properties for $PackageName"
+
+    LogError "Failed to retrive Properties for [ $PackageName ]"
     return $null
 }
 
 # Takes ServiceName and Repo Root Directory
 # Returns important properties for each package in the specified service, or entire repo if the serviceName is not specified
 # Returns an Table of service key to array values of PS Object with properties @ { pkgName, pkgVersion, pkgDirectoryPath, pkgReadMePath, pkgChangeLogPath }
-function Get-AllPkgProperties ([string]$ServiceDirectory = $null)
+function Get-AllPkgProperties ([string]$ServiceDirectoryName = $null)
 {
     $pkgPropsResult = @()
 
-    if ([string]::IsNullOrEmpty($ServiceDirectory))
+    if (Test-Path "Function:Get-AllPackageInfoFromRepo")
     {
-        $searchDir = Join-Path $RepoRoot "sdk"
-        foreach ($dir in (Get-ChildItem $searchDir -Directory))
-        {
-            $serviceDir = Join-Path $searchDir $dir.Name
-
-            if (Test-Path (Join-Path $serviceDir "ci.yml"))
-            {
-                $activePkgList = Get-PkgListFromYml -ciYmlPath (Join-Path $serviceDir "ci.yml")
-                if ($activePkgList -ne $null)
-                {
-                    $pkgPropsResult = Operate-OnPackages -activePkgList $activePkgList -ServiceDirectory $dir.Name -pkgPropsResult $pkgPropsResult
-                }
-            }
-        }
-    } 
+        $pkgPropsResult = Get-AllPackageInfoFromRepo
+    }
     else
     {
-        $serviceDir = Join-Path $RepoRoot "sdk" $ServiceDirectory
-        if (Test-Path (Join-Path $serviceDir "ci.yml"))
+        if ([string]::IsNullOrEmpty($ServiceDirectoryName))
         {
-            $activePkgList = Get-PkgListFromYml -ciYmlPath (Join-Path $serviceDir "ci.yml")
-            if ($activePkgList -ne $null)
+            $searchDir = Join-Path $RepoRoot "sdk"
+            foreach ($dir in (Get-ChildItem $searchDir -Directory))
             {
-                $pkgPropsResult = Operate-OnPackages -activePkgList $activePkgList -ServiceDirectory $ServiceDirectory -pkgPropsResult $pkgPropsResult
+                $serviceDirPath = Join-Path $searchDir $dir.Name
+
+                $pkgPropsResult += Get-PkgPropsForEntireService -serviceDirectoryPath $serviceDirPath `
+                -ServiceDirectoryName $dir.Name
             }
+        } 
+        else
+        {
+            $serviceDirPath = Join-Path $RepoRoot "sdk" $ServiceDirectoryName
```

Comment:
Should we just do the sdk join code inside of the Get-PkgPropsForEntireService? It seems like we just repeat the logic and add an extra parameter to it for no great reason.